app/controllers/api/v1/ownership_certificate_validation_requests_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class OwnershipCertificateValidationRequestsController < Api::V1::ApplicationController
+      skip_before_action :verify_authenticity_token
+      before_action :check_token_and_set_application
+      before_action :set_ownership_certificate_validation_requests, only: %i[index]
+      before_action :set_ownership_certificate_validation_request, only: %i[show update]
+
+      def index
+        respond_to do |format|
+          format.json
+        end
+      end
+
+      def show
+        respond_to do |format|
+          format.json
+        end
+      rescue ActiveRecord::RecordNotFound
+        format.json do
+          render json: {message: "Unable to find ownership certificate validation request with id: #{params[:id]}"},
+            status: :not_found
+        end
+      end
+
+      def update
+        @ownership_certificate_validation_request.update!(ownership_certificate_params.except(:params))
+        @ownership_certificate_validation_request.close!
+        @ownership_certificate_validation_request.create_api_audit!
+        @planning_application.send_update_notification_to_assessor
+        @planning_application.update(valid_ownership_certificate: true)
+
+        if @ownership_certificate_validation_request.approved?
+          OwnershipCertificateCreationService.new(
+            params: ownership_certificate_params[:params], planning_application: @planning_application
+          ).call
+        end
```

Comment:
This feels like a lot of stuff outside a transaction block - there's a bunch of methods there with `!` which suggests that any of them could raise an exception and leave things in an indeterminate state. It might be better to extract this to a separate class which can be tested in isolation and not just in a system test.